fieldsolver/ldz_magnetic_field.cpp

```diff
@@ -231,17 +240,48 @@ void propagateMagneticFieldSimple(
    timer=phiprof::initializeTimer("Compute cells");
    phiprof::start(timer);
    
-   #pragma omp parallel for collapse(3)
+   #pragma omp parallel for collapse(3) schedule(dynamic,1)
    for (int k=0; k<gridDims[2]; k++) {
       for (int j=0; j<gridDims[1]; j++) {
          for (int i=0; i<gridDims[0]; i++) {
 
             // Set the fsgrid rank in the technical grid
+            // FIXME Why is this done repeatedly here? Yann can't remember (20191119).
             technicalGrid.get(i,j,k)->fsGridRank=technicalGrid.getRank();
 
-            if(technicalGrid.get(i,j,k)->sysBoundaryFlag != sysboundarytype::NOT_SYSBOUNDARY) continue;
-            // Propagate B on all local cells:
-            propagateMagneticField(perBGrid, perBDt2Grid, EGrid, EDt2Grid, i, j, k, dt, RKCase);
+            if(technicalGrid.get(i,j,k)->sysBoundaryFlag == sysboundarytype::NOT_SYSBOUNDARY) {
+               // Propagate B on all local cells on all faces:
+               propagateMagneticField(perBGrid, perBDt2Grid, EGrid, EDt2Grid, i, j, k, dt, RKCase, true, true, true);
+            } else {
+               // Easy case: in case we are neighboured by a non-sysboundary cell, we still solve the
+               // fields normally here.
+               cuint sysBoundaryLayer = technicalGrid.get(i,j,k)->sysBoundaryLayer;
+               if(sysBoundaryLayer == 1) {
+                  for (uint component = 0; component < 3; component++) {
```

Comment:
The component loop here could be replaced with:
	bool prop_x = (technicalGrid.get(i-1,j,k)->sysBoundaryFlag == sysboundarytype::NOT_SYSBOUNDARY);
	bool prop_y = (technicalGrid.get(i,j-1,k)->sysBoundaryFlag == sysboundarytype::NOT_SYSBOUNDARY);
	bool prop_z = (technicalGrid.get(i,j,k-1)->sysBoundaryFlag == sysboundarytype::NOT_SYSBOUNDARY);
        if (prop_x || prop_y || prop_z) propagateMagneticField(perBGrid, perBDt2Grid, EGrid, EDt2Grid, i, j, k, dt, RKCase, prop_x, prop_y, prop_z);

Reply:
OK. Won't the compiler make it efficient anyway? But indeed it would be a bit more elegant.

Reply:
Ah but this is actually activating more directions than intended.